packages/bpk-theming/src/BpkThemeProvider.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const uniq = (arr) => {
+  const seen = {};
+  return arr.filter(((item) => {
+    if (seen.hasOwnProperty[item]) {
+      return false;
+    }
+    seen[item] = true;
+    return true;
+  }));
+};
+
+const createStyle = (theme, themeAttributes) => {
+  let style = {};
+  const missingThemeAttributes = [];
+  themeAttributes.forEach((attribute) => {
+    if (theme[attribute]) {
+      const cssName = attribute.replace(/([A-Z])/g, variable => `-${variable.toLowerCase()}`);
+      const value = theme[attribute];
+      style[`--bpk-${cssName}`] = value;
+    } else {
+      missingThemeAttributes.push(attribute);
+    }
+  });
+
+  if (missingThemeAttributes.length > 0) {
+    style = {};
+    console.warn(`BpkThemeProvider: To apply theming, the theme prop must include \`${themeAttributes.join(', ')}\` (missing \`${missingThemeAttributes.join(', ')}\`)`); // eslint-disable-line max-len,no-console
+  }
+  return style;
+};
+
+const BpkThemeProvider = (props) => {
+  const { children, theme, themeAttributes } = props;
+  const dedupedThemeAttributes = uniq(themeAttributes);
```

Comment:
We need to flatten too. Consider

```javascript
import { themeAttributes as buttonThemeAttributes } from 'bpk-component-button';
import { themeAttributes as linkThemeAttributes } from 'bpk-component-link';

() => (
  <ThemeProvider themeAttributes={[ buttonThemeAttributes, linkThemeAttributes ]} theme={ /* ... */}>
    <BpkButton>Search Flights</BpkButton>
    <BpkLink>Click here</BpkLink>
  </ThemeProvider>
);
```

In this case you have an array with two arrays in it. Unless we require the consumer to spread the arrays.

Reply:
In `PropTypes`, it specifies `PropTypes.arrayOf(PropTypes.string)` - I think we should leave this to consumers.

Reply:
cc @matthewdavidson, @matteo-hertel. I think I prefer not forcing people to spread it. If we don't require spreading both spreading and not will work. i.e

```javascript
themeAttributes={[ buttonThemeAttributes, linkThemeAttributes ]}
```
and 

```javascript
themeAttributes={[ ...buttonThemeAttributes, ...linkThemeAttributes ]}
```

will both work

Reply:
Yeah fair enough - that's better for consumers, after all.